src/NServiceBus.FileBasedRouting/XmlRoutingFileParser.cs

```diff
@@ -42,23 +45,32 @@ public IEnumerable<EndpointRoutingConfiguration> Parse(XDocument document)
             return configs;
         }
 
-        static IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
+        IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
         {
             var handles = endpointElement.Element("handles");
 
             var separatelyConfiguredCommands = handles
                 ?.Elements(singular)
                 .Select(e => SelectCommand(e.Attribute("type").Value))
+                .Where(e => e != null)
```

Comment:
it does and that's the point. In a centralized configuration, you have all the routes, even though those messages / types might not be available on that endpoint.

I also thought a `warn` log would be enough.